File: Programmers/c++/두 개 뽑아서 더하기.cpp
/*
프로그래머스 두 개 뽑아서 더하기
숫자 2개를 뽑아서 더한 수들을 오름차순으로 반환.
이중 반복문을 통해 모든 경우의 수를 구하고 map key값의 특성을 이용하여 중복 해결.
map의 key값들을 answer에 입력.
*/
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(vector<int> numbers) {
	vector<int> answer;
	map<int, int> m;
	for (int i = 0; i<numbers.size() - 1; i++) {
		for (int j = i + 1; j<numbers.size(); j++) {
			int a = numbers[i] + numbers[j];
			if (m.find(a) == m.end()) {
				m.insert(make_pair(a, 1));
			}
		}
	}
	for (auto i = m.begin(); i != m.end(); i++) {
		answer.push_back(i->first);
	}
	return answer;
}

File: programming/Programmers/javascript/두 개 뽑아서 더하기.js
/*
프로그래머스 두 개 뽑아서 더하기
숫자 2개를 뽑아서 더한 수들을 오름차순으로 반환.
중복의 숫자는 제외하고 answer에 삽입
마지막에 sort함수를 통해 오름차순 정렬.
*/
function solution(numbers) {
    var answer = [];
    for(var i=0; i<numbers.length-1; i++){
        for(var j=i+1; j<numbers.length; j++){
            var  n=numbers[i]+numbers[j];
            if(answer.indexOf(n)==-1){
                answer.push(n);
            }
        }
    }
    answer.sort(function(a, b)  {
        return a - b;
    });
    return answer;
}
